File: LegacyOctave/formantcos.m
function [formant] = formantcos(TAILLE, SURECHANTILLONNAGE)
  PI = 3.1415926535;
  formant = zeros(1,TAILLE);
  t = ([1:1:TAILLE] - floor((TAILLE+1)/2)) / SURECHANTILLONNAGE;
  for i=1:TAILLE
    if (t(i)*t(i)) == (1/16)
      formant(i) = PI / 4;
    else
      formant(i) = cos(2*PI*t(i)) ./ (1 - 16 * t(i) * t(i));
    end;
  end
end

File: formantcos.jl
function formantcos(taille, surechantillonnage)
    pi = 3.1415926535
    formant = zeros(taille)
    t = collect(1:taille) - Int((taille+1) / 2) 
    t = t / surechantillonnage
    for i = collect(1:1:taille)
        if t[i]^2 == (1/16)
            formant[i] = pi/4
        else
            formant[i] = cospi(2*t[i]) / (1 - 16 * t[i]^2)
        end
    end
    return formant
end
